### ISSUE TITLE
this.props in getOptimisticResponse is cached value

### ISSUE THREAD

**Issue by walfly** [facebook/relay#1394]
I'm trying to optimistically update an item description. I'm passing in an item with an updated description to the mutation, but inside  the getOptimisticResponse this.props.item has the old value of the description.  The only reference to the passed in props is _unresolvedProps.  Is there a way to optimistically set the value of the item.desciption to the new description without accessing private properties or extending the constructor?

**Comment by Gregoor** [comment]
Mutations can have fragments, you could have a fragment on the item, with the description. Then you can access it via `this.item.description`. For more info on it, look here: https://facebook.github.io/relay/docs/guides-mutations.html#a-complete-example

**Comment by wincent** [comment]
Thanks for filing this issue @walfly and sorry for not responding sooner. I'm going to optimistically close this on the assumption that @Gregoor's tip was enough to unblock you and get you moving forward. Please feel free to comment again or open a new issue if you think you've found a bug. If you have a doubt a about usage (ie. a question, and you don't think it's a bug), I'd encourage you to post to Stack Overflow under the tag #relayjs; we generally try to keep the signal strong in the GitHub issue tracker (for issues and features), and think you're more likely to get a timely response if you use a purpose-built Q&A forum like Stack Overflow. If you do post a question there about this, please share a link to it back here so that anybody else stumbling across this can benefit from any answers there too.

https://stackoverflow.com/questions/ask?tags=relayjs
